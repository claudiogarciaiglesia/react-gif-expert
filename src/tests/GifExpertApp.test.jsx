import { fireEvent, getAllByText, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../GifExpertApp"

describe('<GiftExpertApp /> testing', () => {

    const inputTestStrig = 'this is a test';

    test('should match snapshot', () => {

        const { container } = render(<GifExpertApp />);
        // screen.debug();

        expect(container).toMatchSnapshot();

    });

    test('should update input value when writing', () => {

        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');


        fireEvent.input(input, { target: { value: inputTestStrig } });
        // screen.debug();

        expect(input.value).toBe(inputTestStrig);

    });

    test('should add unique category when submitting form', () => {

        render(<GifExpertApp />);
        const form = screen.getByRole('form');
        const input = screen.getByRole('textbox');

        // Try to submit form with same input
        fireEvent.input(input, { target: { value: inputTestStrig } });
        fireEvent.submit(form);
        fireEvent.input(input, { target: { value: inputTestStrig } });
        fireEvent.submit(form);

        // Create different input
        fireEvent.input(input, { target: { value: inputTestStrig + 1 } });
        fireEvent.submit(form);

        expect(screen.getAllByText(inputTestStrig).length).toBe(1);
        expect(screen.getAllByText(inputTestStrig + 1).length).toBe(1);
        expect(screen.getAllByLabelText('gif-grid').length).toBe(3);


    });




});