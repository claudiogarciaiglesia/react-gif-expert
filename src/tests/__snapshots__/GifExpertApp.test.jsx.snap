// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftExpertApp /> testing should match snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search for gifs"
      type="text"
      value=""
    />
  </form>
  <h3
    aria-label="gif-grid"
  >
    One Punch
  </h3>
  <h3>
    Cargando...
  </h3>
  <div
    class="card-grid"
  />
</div>
`;
